RecoTracker/FinalTrackSelectors/plugins/TrackTfClassifier.cc

```diff
@@ -0,0 +1,123 @@
+#include "RecoTracker/FinalTrackSelectors/interface/TrackMVAClassifier.h"
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+#include "getBestVertex.h"
+
+#include "TrackingTools/Records/interface/TfGraphRecord.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+#include "RecoTracker/FinalTrackSelectors/interface/TfGraphDefWrapper.h"
+
+struct TfDnnCache {
+  TfDnnCache() : session_(nullptr) {}
+  tensorflow::Session* session_;
+};
+
+namespace {
+  struct TfDnn {
+    TfDnn(const edm::ParameterSet& cfg)
+        : tfDnnLabel_(cfg.getParameter<std::string>("tfDnnLabel"))
+
+    {}
+    TfDnnCache* cache_ = new TfDnnCache();
+
+    ~TfDnn() {
+      if (cache_->session_) {
+        tensorflow::closeSession(cache_->session_);
+        delete cache_;
+      }
+    }
+
+    static const char* name() { return "TrackTfClassifier"; }
+
+    static void fillDescriptions(edm::ParameterSetDescription& desc) {
+      desc.add<std::string>("tfDnnLabel", "trackSelectionTf");
+    }
+
+    void beginStream() {}
+
+    void initEvent(const edm::EventSetup& es) {
+      if (!cache_->session_) {
+        edm::ESHandle<TfGraphDefWrapper> tfDnnHandle;
+        es.get<TfGraphRecord>().get(tfDnnLabel_, tfDnnHandle);
```

Comment:
`ESHandle::get()` is deprecated, use `ESGetToken` to get data from the `EventSetup`: https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_a_Record

you can see this PR for an example: https://github.com/cms-sw/cmssw/pull/32246/files

Reply:
Hello, it seems like I need to include the new header file for esConsumes(), but I am not sure which header is supposed to used in here.

Reply:
esConsumes needs to come from the EDProducer, or passed via ConsumesCollector. This will require a few changes to the other Track MVA classes. So that this PR here doesn't get stuck or expand in scope too much, let's defer the ESGetToken update for a follow-up.